force-app/main/default/classes/DMLOperations.cls

```diff
@@ -17,204 +17,337 @@
+   * @param name The name of the account to be created.
+   * @param industry The industry of the account to be created.
+   */
+  public static void createAccount(String name, String industry) {
+    // Create a new Account and set the fields inline
+    Account account = new Account();
+    account.Name = name;
+    account.Industry = industry;
+
+    // Insert the Account
+    insert account;
+  }
+
+  /**
+   * Question 3
+   * Create a new Contact and link it to the Account id provided then inserts it into the database.
+   * @param accountId The Id of the Account to link the Contact to.
+   * @return The Id of the newly created Contact.
+   */
+  public static Id insertNewContact(Id accountId) {
+    // Create a new Contact instance
+    Contact contact = new Contact();
+
+    // Set some fields on the Contact
+    // Set the AccountId field to the given accountId
+    contact.AccountId = accountId;
+    contact.FirstName = 'Musa';
+    contact.LastName = 'Sannoh';
+
+    // Insert the Contact into the database
+    insert contact;
+
+    // Return the Id of the newly created Contact
+    return contact.Id;
+  }
+
+  /**
+   * Question 4
+   * Update the last name of a Contact in the database.
+   * Resource: https://developer.salesforce.com/docs/atlas.en-us.soql_sosl.meta/soql_sosl/sforce_api_calls_soql.htm
+   * @param contactId The Id of the Contact to update.
+   * @param newLastName The new last name for the Contact.
+   */
+  public static void updateContactLastName(Id contactId, String newLastName) {
+    // Retrieve the Contact from the database
+    Contact contact = [SELECT Id FROM Contact WHERE Id = :contactId];
+
+    // Update the Contact's last name
+    contact.LastName = newLastName;
+    // Update the Contact in the database
+    update contact;
+  }
+
+  /**
+   * Question 5
+   * Update the stage of an Opportunity in the database.
+   * Resource: https://developer.salesforce.com/docs/atlas.en-us.soql_sosl.meta/soql_sosl/sforce_api_calls_soql.htm
+   * @param oppId The Id of the Opportunity to update.
+   * @param newStage The new stage for the Opportunity.
+   */
+  public static void updateOpportunityStage(Id oppId, String newStage) {
+    // Retrieve the Opportunity from the database
+    Opportunity opp = [SELECT Id FROM Opportunity WHERE Id = :oppId];
+
+    // Update the Opportunity's stage
+    opp.StageName = newStage;
+
+    // Update the Opportunity in the database
+    update opp;
+  }
+
+  /**
+   * Question 6
+   * Update the Account's Name and Industry given an Account Id.
+   * Resource: https://developer.salesforce.com/docs/atlas.en-us.soql_sosl.meta/soql_sosl/sforce_api_calls_soql.htm
+   * @param accountId The Id of the account to be updated.
+   * @param newName The new name for the account.
+   * @param newIndustry The new industry for the account.
+   */
+  public static void updateAccountFields(
+    Id accountId,
+    String newName,
+    String newIndustry
+  ) {
```

Comment:
Very elegant but try to stick to doing things one way only so you don't confuse the reader or yourself later.